cli/config/configfile/file.go

```diff
@@ -351,3 +351,26 @@ func (configFile *ConfigFile) SetPluginConfig(pluginname, option, value string)
 		delete(configFile.Plugins, pluginname)
 	}
 }
+
+// GetFeature retrieves the given key from the config features map.
+func (configFile *ConfigFile) GetFeature(key string) (string, bool) {
+	if configFile.Features == nil {
```

Comment:
Will we need any synchronisation around these?

Also the `Features` being both exported _and_ having accessors starts to feel "wrong" (admitted, we already have a similar situation for the `Plugins` field above).

Reply:
Oh! I forgot to hit the submit button before I got up for lunch 🙃 I was about to post a comment asking about whether this made sense or not — I came across the Plugins one while working downstream.

Synchronization would benefit us here, although it's useless unless we implement an actual file lock to prevent concurrent access by different processes, which I think is a bigger concern.

Reply:
Yeah, this is gonna be more of a problem if more plugins start to manipulate these out of band; more so if it's stored in memory, and now possibly memory state no longer reflecting actual state of the file, think;

- cli loads config in memory
- calls plugin -> which may manipulate the config
- now cli's config in memory is out of sync
- but cli could write to disk (reverting the changes made by the plugin)

I haven't checked the code locally, but if `*ConfigFile` here is the struct that includes the `json:` labels, then I'm guessing these fields were exported _because of that_ (i.e. (un)marshaling requires them to be exported), so probably we can't change that immediately, unless we define separate structs/types (separate types for (un)marshaling than for "in-memory" state; the latter would have un-exported fields and accessors).

Reply:
Hmm, I mostly agree. I think ultimately this is something that will only be "properly" solved by a) having the plugins not change the config directly, but communicate changes over the socket to the CLI and have the CLI process make those changes, and b) implementing a file lock so other processes don't touch the config at the same time.

However, for right now, barring concurrent process access, I think this:

> cli loads config in memory
> calls plugin -> which may manipulate the config
> now cli's config in memory is out of sync
> but cli could write to disk (reverting the changes made by the plugin)

Isn't a huge problem since we can know when we're writing the file to disk from the CLI process and can try to make sure we don't do it while the CLI is calling a plugin.

Reply:
I can shelve this PR until we have file locking in place, or we can keep it (even in a somewhat incomplete state) to add synchronization, although I don't think that will bring us as much value right now.